### PR TITLE
[feat] Add create_metric_task for easy timing metrics for asyncio tasks.

### LINKED ISSUE #28: Feature suggestion - timed task
Hi,
In our use case we've implemented a function for timing asyncio tasks - essentially we have a wrapper function for create_task that adds done callback that sends the metric of duration, if exceeds certain threshold.
I wonder if it's something that would be welcome in this library, if so I'll create a PR!

### PR DESCRIPTION
**Pull request by aviramha**
Closes #28

### CODE REVIEW

**Comment by aviramha** [comment]
@Gr1N Awaiting for feedback!

**Comment by Gr1N** [comment]
Hi, sorry for the delay. I will take a look today.

**Comment by aviramha** [comment]
Any news ?

**Comment by Gr1N** [comment]
Sorry for the delay, this month a lot of work and have no time for other activities. I Will try to take a detailed look tomorrow. Overall LGTM but I have a few questions but first want to look a bit deeper.

File: aiodogstatsd/client.py [hunk 1]
```diff
@@ -272,6 +274,32 @@ def timeit(
             if not threshold_ms or value > threshold_ms:
                 self.timing(name, value=int(value), tags=tags, sample_rate=sample_rate)
 
+    def create_metric_task(
```
Comment (Gr1N) [inline aiodogstatsd/client.py:277]:
```suggestion
    def timeit_task(
```

File: aiodogstatsd/client.py [hunk 2]
```diff
@@ -272,6 +274,32 @@ def timeit(
...
+        self,
+        coro: Awaitable[_T],
+        name: typedefs.MName,
+        *,
+        tags: Optional[typedefs.MTags] = None,
+        sample_rate: Optional[typedefs.MSampleRate] = None,
+        threshold_ms: Optional[typedefs.MValue] = None,
+    ) -> "asyncio.Task[_T]":
```
Comment (Gr1N) [inline aiodogstatsd/client.py:285]:
```suggestion
    ) -> asyncio.Task[_T]:
```

File: aiodogstatsd/client.py [hunk 3]
```diff
@@ -272,6 +274,32 @@ def timeit(
...
+        """
+        Creates a task and returns it, adds a done callback for
+            sending time metric if done and exceeds threshold.
+        """
+        loop = get_event_loop()
+        start_time = loop.time()
```
Comment (Gr1N) [inline aiodogstatsd/client.py:291]:
```suggestion
        started_at = loop.time()
```

File: aiodogstatsd/client.py [hunk 4]
```diff
@@ -272,6 +274,32 @@ def timeit(
...
+
+        def _conditional_time(_):
```
Comment (Gr1N) [inline aiodogstatsd/client.py:293]:
```suggestion
        def _callback(_: Any) -> None:
```

**Comment by Gr1N** [comment]
And please update https://github.com/Gr1N/aiodogstatsd/blob/master/CHANGELOG.md

File: aiodogstatsd/client.py (hunk 2 above)
Comment (aviramha) [reply aiodogstatsd/client.py:285]: I'm pretty sure this doesn't work at runtime due to `asyncio.Task` not being a generic in 3.7/3.8

Comment (aviramha) [reply aiodogstatsd/client.py:285]:
```
tests/test_protocol.py:3: in <module>
    from aiodogstatsd import protocol, typedefs
aiodogstatsd/__init__.py:1: in <module>
    from .client import Client
aiodogstatsd/client.py:15: in <module>
    class Client:
aiodogstatsd/client.py:285: in Client
    ) -> asyncio.Task[_T]:
E   TypeError: 'type' object is not subscriptable
```

**Comment by aviramha** [comment]
Fixed all comments, regarding `asyncio.Task[_T]` see my reply above :)

File: docs/usage.md [hunk 5]
```diff
@@ -93,9 +93,19 @@ client.timing("query.time", value=0.5)
...
 
-Context manager for easily timing methods, optionally settings `tags` and a `sample_rate`.
+Context manager for easily timing methods, optionally settings `tags`, `sample_rate` and `threshold_ms`.
 
 ```python
 with client.timeit("query.time"):
     ...
 ```
+
+### create_metric_task
```
Comment (Gr1N) [inline docs/usage.md:103]: Outdated a bit, please update and I'm going to merge PR

File: examples/create_metric_task.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+
+async def main():
+    client = aiodogstatsd.Client(
+        host="0.0.0.0", port=9125, constant_tags={"whoami": "I am Batman!"}
+    )
+    await client.connect()
+
+    for _ in range(5000):
+        await client.create_metric_task(do_something(), "task_finished")
```
Comment (Gr1N) [inline examples/create_metric_task.py:17]: Same here

File: examples/create_metric_task.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import asyncio
+
+import aiodogstatsd
+
+
+async def do_something():
+    asyncio.sleep(1)
```
Comment (Gr1N) [inline examples/create_metric_task.py:7]:
```suggestion
    await asyncio.sleep(1)
```

File: docs/usage.md [hunk 8]
```diff
@@ -93,9 +93,19 @@ client.timing("query.time", value=0.5)
...
+
+Wrapper for `asyncio.create_task` that creates a task from a given `Awaitable` and sends timing metric of it's duration.
+
+```python
+async def do_something():
+    asyncio.sleep(1.0)
```
Comment (Gr1N) [inline docs/usage.md:109]:
```suggestion
    await asyncio.sleep(1.0)
```

**Comment by aviramha** [comment]
@Gr1N done

**Comment by Gr1N** [comment]
Thanks!

**Comment by aviramha** [comment]
@Gr1N When do you plan to release the next version? I'd like to consume it via PyPI ASAP 👯

**Comment by Gr1N** [comment]
@aviramha enjoy! https://pypi.org/project/aiodogstatsd/0.15.0/

**Pull request merged**